src/Component/ChatBox/ChatBox.tsx

```
@@ -0,0 +1,157 @@
+
+  useEffect(() => {
+    if (messageArray.length === 0) {
+      botMessage();
+    } else {
+      scrollToBottom();
+    }
+  }, [setMessageArray.length]);
+
+  const sendMessage = (message: any) => {
+    if (!message) {
+      return;
+    }
+
+    const messageObj: Actor = {
+      messageId: 1,
+      id: "user",
+      displayName: "John Doe",
+      align: "right",
+      message,
+      delay: 0,
+      input: {
+        type: "answer",
+      },
+    };
+
+    const arr = messageArray;
+    arr.push(messageObj);
+    setMessageArray(arr);
+    botMessage();
+  };
+
+  const botMessage = () => {
+    setLoading(true);
+    const botObj: Actor = BOT_QUESTIONS[questionIndex];
+
+    setTimeout(() => {
+      const newArr = messageArray;
+      newArr.push(botObj);
+      setMessageArray(newArr);
+      setInput(botObj.input);
+      setQuestionIndex(questionIndex + 1);
+      setLoading(false);
+    }, botObj.delay);
+  };
+
+  const scrollToBottom = () => {
+    if (messagesWrapperRef.current) {
+      const node = messagesWrapperRef.current;
+      node.scrollTop = node.scrollHeight;
+    }
+  };
+
+  let inputComponent;
+  switch (input.type) {
+    case "text":
+      inputComponent = (
+        <TextInput name={input.name} sendMessage={sendMessage} />
+      );
+      break;
+
+    case "textarea":
+      inputComponent = (
+        <TextAreaInput name={input.name} sendMessage={sendMessage} />
+      );
+      break;
+
+    case "radio":
+      inputComponent = (
+        <RadioInput
+          name={input.name}
+          sendMessage={sendMessage}
+          options={input.options}
+        />
+      );
+      break;
+
+    case "checkbox":
+      inputComponent = (
+        <CheckboxInput
+          name={input.name}
+          sendMessage={sendMessage}
+          options={input.options}
+        />
+      );
+      break;
+
+    case "empty":
+      inputComponent = <div />;
+      break;
+
+    default:
+      inputComponent = (
```

should raise an error..